<?xml version="1.0" ?>
<rml version="1.4">
  <world name="default">
    <include>
      <uri>model://ground_plane</uri>
    </include>
    <include>
      <uri>model://sun</uri>
    </include>

    <!-- test contributed by jesper -->
    <model name="boxes">
      <pose>0 0 0 0 0 1.57</pose>
      <link name="link1">
        <collision name="collision">
          <pose>0.5 0 0.5 0 0 0</pose>
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
          <surface>
            <contact>
              <ode>
                <min_depth>0.01</min_depth>
              </ode>
            </contact>
            <friction>
              <ode>
                <mu>1</mu>
                <mu2>1</mu2>
              </ode>
            </friction>
          </surface>
        </collision>
        <visual name="visual">
          <pose>0.5 0 0.5 0 0 0</pose>
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </visual>
        <inertial>
          <mass>100.0</mass>
          <pose>0.5 0 0.5 0 0 0</pose>
          <inertia>
            <ixx>0.1</ixx>
            <iyy>0.1</iyy>
            <izz>0.1</izz>
            <ixy>0.1</ixy>
            <ixz>0.1</ixz>
            <iyz>0.1</iyz>
          </inertia>
        </inertial>
      </link>
      <link name="link2">
        <pose>1.0 0 1.0 0 0 0</pose>
        <collision name="collision">
          <pose>0.5 0 0.5 0 0 0</pose>
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </collision>
        <visual name="visual">
          <pose>0.5 0 0.5 0 0 0</pose>
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </visual>
        <inertial>
          <mass>5.0</mass>
          <pose>0.5 0 0.5 0 0 0</pose>
          <inertia>
            <ixx>0.1</ixx>
            <iyy>0.1</iyy>
            <izz>0.1</izz>
            <ixy>0.1</ixy>
            <ixz>0.1</ixz>
            <iyz>0.1</iyz>
          </inertia>
        </inertial>
      </link>
      <link name="link3">
        <pose>2.0 0 2.0 0 0 0</pose>
        <collision name="collision">
          <pose>0.5 0 0.5 0 0 0</pose>
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </collision>
        <visual name="visual">
          <pose>0.5 0 0.5 0 0 0</pose>
          <geometry>
            <box>
              <size>1 1 1</size>
            </box>
          </geometry>
        </visual>
        <inertial>
          <mass>1.0</mass>
          <pose>0.5 0.5 0.5 0 0 0</pose>
          <inertia>
            <ixx>0.1</ixx>
            <iyy>0.1</iyy>
            <izz>0.1</izz>
            <ixy>0.1</ixy>
            <ixz>0.1</ixz>
            <iyz>0.1</iyz>
          </inertia>
        </inertial>
      </link>
      <joint name="joint1" type="revolute">
        <physics>
          <ode>
            <provide_feedback>true</provide_feedback>
            <cfm_damping>true</cfm_damping>
          </ode>
        </physics>
        <axis>
          <xyz>0 1 0</xyz>
          <dynamics>
            <damping>100.000000</damping>
          </dynamics>
        </axis>
        <parent>link1</parent>
        <child>link2</child>

        <sensor name="force_torque" type="force_torque">
          <always_on>true</always_on>
          <visualize>true</visualize>
          <update_rate>30</update_rate>
        </sensor>

      </joint>
      <joint name="joint2" type="revolute">
        <physics>
          <ode>
            <provide_feedback>true</provide_feedback>
            <cfm_damping>true</cfm_damping>
          </ode>
        </physics>
        <axis>
          <xyz>0 1 0</xyz>
          <dynamics>
            <damping>30.000000</damping>
          </dynamics>
        </axis>
        <parent>link2</parent>
        <child>link3</child>
        <sensor name="force_torque" type="force_torque">
          <always_on>true</always_on>
          <visualize>true</visualize>
          <update_rate>30</update_rate>
        </sensor>
      </joint>
    </model>
  </world>
</rml>
